PhysicsTools/TensorFlow/interface/Tensor.h

```diff
@@ -0,0 +1,450 @@
+        getPtrVector<T>(axis, a, b, c, v);
+        return v;
+    }
+
+    // sets the values of a tensor along an axis for a fixed position on the other axes
+    template <typename T>
+    void setVectorAtPos(int axis, Shape* pos, std::vector<T>& v);
+
+    // sets the values of a rank 1 tensor
+    template <typename T>
+    inline void setVector(std::vector<T>& v)
+    {
+        assertRank(1);
+        setVectorAtPos<T>(0, 0, v);
+    }
+
+    // sets the values of a rank 2 tensor along an axis for a fixed position on the other axis
+    template <typename T>
+    inline void setVector(int axis, Shape a, std::vector<T>& v)
+    {
+        assertRank(2);
+        Shape pos[1] = { a };
+        setVectorAtPos<T>(axis, pos, v);
+    }
+
+    // sets the values of a rank 3 tensor along an axis for a fixed position on the other axes
+    template <typename T>
+    inline void setVector(int axis, Shape a, Shape b, std::vector<T>& v)
+    {
+        assertRank(3);
+        Shape pos[2] = { a, b };
+        setVectorAtPos<T>(axis, pos, v);
+    }
+
+    // sets the values of a rank 4 tensor along an axis for a fixed position on the other axes
+    template <typename T>
+    inline void setVector(int axis, Shape a, Shape b, Shape c, std::vector<T>& v)
+    {
+        assertRank(4);
+        Shape pos[3] = { a, b, c };
+        setVectorAtPos<T>(axis, pos, v);
+    }
+
+    // sets n elements to a constant value starting from a certain position, negative n means all
+    template <typename T>
+    void fillValuesAtPos(T v, int n, Shape* pos);
+
+    // sets n elements of a rank 1 tensor to a constant value starting from a certain position,
+    // negative n means all
+    template <typename T>
+    inline void fillValues(T v, int n, Shape i)
+    {
+        assertRank(1);
+        Shape pos[1] = { i };
+        fillValuesAtPos<T>(v, n, pos);
+    }
+
+    // sets n elements of a rank 2 tensor to a constant value starting from a certain position,
+    // negative n means all
+    template <typename T>
+    inline void fillValues(T v, int n, Shape i, Shape j)
+    {
+        assertRank(2);
+        Shape pos[2] = { i, j };
+        fillValuesAtPos<T>(v, n, pos);
+    }
+
+    // sets n elements of a rank 3 tensor to a constant value starting from a certain position,
+    // negative n means all
+    template <typename T>
+    inline void fillValues(T v, int n, Shape i, Shape j, Shape k)
+    {
+        assertRank(3);
+        Shape pos[3] = { i, j, k };
+        fillValuesAtPos<T>(v, n, pos);
+    }
+
+    // sets n elements of a rank 4 tensor to a constant value starting from a certain position,
+    // negative n means all
+    template <typename T>
+    inline void fillValues(T v, int n, Shape i, Shape j, Shape k, Shape l)
+    {
+        assertRank(4);
+        Shape pos[4] = { i, j, k, l };
+        fillValuesAtPos<T>(v, n, pos);
+    }
+
+private:
+    // the internal tensorflow tensor object
+    TF_Tensor* tf_tensor;
```

Comment:
it looks like the Tensor class has the ownership.
Can this be a unique_ptr instead of plain pointer?
Same q for Graph etc